apps/backend/db_patches/0201_AddUniqueConstraintToOidcSub.sql

```
@@ -0,0 +1,35 @@
+
+DO
+$$
+BEGIN
+    IF register_patch('AddUniqueConstraintToOidcSub.sql', 'yoganandanpandiyan', 'Unique constraint', '2025-10-13') THEN
+        BEGIN
+            -- Clean up empty string values in oidc_sub column by setting them to NULL
+            -- This prevents empty strings from conflicting with the unique constraint
+            UPDATE users
+            SET oidc_sub = NULL
+            WHERE oidc_sub = '';
+
+            -- Handle duplicate oidc_sub values by appending a unique suffix. ex., '_1', '_2', etc.
```

I’m a little bit concerned about this. We silently mutate rows, can’t we just let the update fail and manually fix the rows that have the same values?
Instead of trying to be smart, we could select the duplicate values and, if any are found, raise a clear error indicating where the duplicates are and that they need to be fixed.

---

Sounds like a plan